### ISSUE TITLE
method call precedence in math

### ISSUE THREAD

**Issue by damaxwell** [typst/typst#2015]
### Description

Here's a subtle difference between math and code:
```
#let foo(..args) = $args.pos().join(",")$

$ foo(a,b,c) $
```

<img width="257" alt="image" src="https://github.com/typst/typst/assets/918465/4ad3c257-26ac-4ce6-a6f6-905a6b37e230">

The method call `.join` is interrupted by the `()` and you have two pieces, `args.pos()` and `.join(,)`.  Not sure if this is the desired behavior.

### Typst version

- [X] I am using the latest version of Typst

**Comment by frozolotl** [comment]
It's actually not related to precedence. It's just that in math mode, a field access (which does include function calls) is simply a dot-separated list of identifiers.
So, `foo.bar.baz` is a valid field access, but `foo.bar().baz` is not valid because `bar()` is not an identifier.

This issue is still very valid though, I just wanted to explain the cause.

**Comment by damaxwell** [comment]
Yep, I know the cause in the parser but am unsure if it is intentional.  Maybe associativity was really the word I was looking for (the dot is behaving like it is right associative). I'll clarify the title to something else, regardless.

**Comment by laurmaedje** [comment]
For context: The only reason the grammar allows method calls in math in the first place is for accents like `$arrow.l(x)$`.

**Comment by damaxwell** [comment]
I see.  And as a side effect, this is why
```
#let foo = ([a], [b], [c])
$ foo.join(",") $
```
works just fine.

**Comment by laurmaedje** [comment]
The math syntax already has enough intricacy as is, so I'd rather not support method chains like this, which have very limited use in math.
